### PR TITLE
feat(buttons): CSS-in-JS refactor

### LINKED ISSUE #174: Give styled buttons access to keyboardFocused state
## Expectations

The Garden Button displays a focus shadow when it receives focus as a result of a keyboard interaction, but not when it receives focus as a result of a mouse interaction.

This is accomplished using the `KeyboardFocusContainer`, which provides a `keyboardFocused` flag to the render prop.

It is often necessary to add styling to the Garden button, for example to change its background color. In this scenario, it is also necessary to change the color of the focus shadow. One could do this with:

```js
const StyledButton = styled(Button)`
  background-color: red;
  :focus {
    box-shadow: 0 0 0 3px red;
  }
`;
```

but `:focus` does not have the same behaviour the actual button, which adds the `is-focused` class on `keyboardFocused || props.focused`.

There should be a way for styled buttons to override styling for specific button states, without having to reimplement the logic of how it gets into that state.

## Reality

There isn't.

## Suggestion

In Button:

```js
class Button {
  ...
  static classNames = ButtonStyles;
}
```

In StyledButton:

```js
const StyledButton = styled(Button)`
  background-color: red;
  &.${Button.classNames['is-focused']} {
     box-shadow: 0 0 0 3px red;
  }
`;
```

As CSS Components is also an open source project, class names like 'is-focused' are already a documented part of the Garden public API.

This approach is generic enough to handle all such use cases (other states and other components).

###### Fine Print

* Component: Buttons, but maybe others?
* Browsers: all

### PR DESCRIPTION
**Pull request by jzempel**
- [x] **BREAKING CHANGE?** 

## Description

Key elements of this PR include:

- [BREAKING CHANGES](https://github.com/zendeskgarden/react-components/pull/399/files#diff-88f886ae93cea50aceea3844633ae74d)
  - no more `styles.css` dist
  - removed `ButtonGroupView` export
  - removed `Icon` export
- Restructured source
  - `examples` – directory for StyleGuidist markdown
  - `src/components` – high-level package components
  - `src/styled` – styled-components variants (formerly `views`)
- All button CSS contained within components and themable! (no more CSS module dependencies)
  - button styling is enhanced per design requirements /cc @ginnywood 
- Enhanced `getColor` to make shade adjustments to non-palette `hue` values

## Detail

closes #174 via `.focus-visible` class styling that can be targeted for `:focus` CSS overrides (although updated (theming)[https://github.com/zendeskgarden/react-components/pull/381] should be considered instead).

## Checklist

- [x] :ok_hand: design updates are Garden Designer approved (add the
      designer as a reviewer)
- [ ] :nail_care: ~view component styling is based on a Garden CSS
      component~
- [x] :globe_with_meridians: Styleguidist demo is up-to-date (`yarn start`)
- [x] :arrow_left: renders as expected with reversed (RTL) direction
- [x] :wheelchair: analyzed via [axe](https://www.deque.com/axe/) and evaluated using VoiceOver
- [x] :guardsman: includes new unit tests
- [x] :memo: tested in Chrome, Firefox, Safari, Edge, and IE11

### CODE REVIEW

File: BREAKING_CHANGES.md [hunk 1]
```diff
@@ -1,5 +1,16 @@
...
 
+## @zendeskgarden/react-buttons
+
+- no longer packages a `styles.css` dist; CSS is self-contained
+- removed `ButtonGroupView`; use new `SplitButton` component instead
+- removed `Icon`; just use `IconButton` with a `rotated` prop
+
+## @zendeskgarden/react-loaders
+
+- `<Dots />` loader no longer has the `velocity` prop
```
Comment (ryanseddon) [inline BREAKING_CHANGES.md]: Do you need to rebase next this should be merged already?

**Comment by jzempel** [comment]
@ryanseddon looks like this PR is weighing-in at about 59.3K vs. 60.3K for the combined (45.8 `index.js` + 14.5 `styles.css`) dist https://unpkg.com/@zendeskgarden/react-buttons/. So not a massive size savings – but with a substantially improved theming experience.

**Comment by austin94** [comment]
> So not a massive size savings – but with a substantially improved theming experience

The size savings will become more visible for areas that are re-exporting our components. There are some areas where this will avoid duplicated CSS sheets.

Also, once we get ESM bundles added the tree-shaking aspect will help some.

File: utils/test/svg-mock.js [hunk 2]
```diff
@@ -5,8 +5,10 @@
  * found at http://www.apache.org/licenses/LICENSE-2.0.
  */
 
-import React from 'react';
+import styled from 'styled-components';
 
-const MockSVG = props => <span {...props}>SVG</span>;
+const MockSVG = styled.svg.attrs(() => ({
```
Comment (austin94) [inline utils/test/svg-mock.js]: Why is `styled-components` necessary for this? Could we just use a standard component. Like:

```jsx
const MockSVG = props => <svg xmlns="http://www.w3.org/2000/svg" {...props} />;
```

File: packages/typography/src/views/lists/OrderedListItem.js [hunk 3]
```diff
@@ -31,7 +31,7 @@ const StyledOrderedListItemContent = styled(MD)`
   ${props => listItemContentCSS(props)};
 `;
 
-const OrderedListItem = ({ children, ...props }) => {
+const OrderedListItem = ({ children }) => {
```
[the remark this answers is not in the document]
Comment (jzempel) [reply packages/typography/src/views/lists/OrderedListItem.js]: I'm addressing this in the upcoming `typography` PR: https://github.com/zendeskgarden/react-components/compare/next...jzempel/styled-typography#diff-df84cee4e13dc0397a975151f9b34145R17

File: utils/test/svg-mock.js (hunk 2 above)
Comment (jzempel) [reply utils/test/svg-mock.js]: Yeah, had it that way originally and then decided to be fancy. Reverting.

File: packages/typography/src/views/lists/UnorderedListItem.js [hunk 4]
```diff
@@ -26,7 +26,7 @@ const StyledUnorderedListItemContent = styled(MD)`
   ${props => listItemContentCSS(props)};
 `;
 
-const UnorderedListItem = ({ children, ...props }) => {
+const UnorderedListItem = ({ children }) => {
```
[the remark this answers is not in the document]
Comment (jzempel) [reply packages/typography/src/views/lists/UnorderedListItem.js]: I'm addressing this in the upcoming `typography` PR: https://github.com/zendeskgarden/react-components/compare/next...jzempel/styled-typography#diff-b6aaaaa36f05ab8393bcd03def2f60b7R17

**Pull request merged**